plugin.py

```diff
@@ -18,22 +17,38 @@ def _flow_config_path(path: str) -> str:
     return join(path, ".flowconfig")
 
 
+def _find_flow_root_path(directory: str) -> Optional[str]:
+    if exists(_flow_config_path(directory)):
+        return directory
+    elif directory == dirname(directory):
+        return None
+    else:
+        return _find_flow_root_path(dirname(directory))
+
+
 class Flow(AbstractPlugin):
     @classmethod
     def name(cls) -> str:
         return cls.__name__.lower()
 
     @classmethod
-    def can_start(cls, _: sublime.Window, __: sublime.View,
-        folders: List[WorkspaceFolder], configuration: ClientConfig
+    def can_start(
+        cls,
+        _: sublime.Window,
+        initiating_view: sublime.View,
+        folders: List[WorkspaceFolder],
+        configuration: ClientConfig,
     ) -> Optional[str]:
-        if not folders:
-            return "need a folder"
-        path = folders[0].path
-        if not exists(_flow_config_path(path)):
-            return "no .flowconfig present in {}".format(path)
+        path = _find_flow_root_path(initiating_view.file_name())
+
+        if not path:
```

Comment:
Since you've removed the requirement that there are workspace folders, does the server work correctly in that case? As in, when a single file is opened in a window and there are no folders in the side bar? Many servers don't work in that case.

Reply:
I forgot to follow up on that but some servers do have problems with that since they consider the root project path (which is the directory of the opened file in case of opening a single file) to be the only location they consider when looking for the configuration (that would be `.flowconfig` in this case) file.